Singularity containers with cloud build

wkharold opened the pull request:
This pull request contains a working version of a community solution that demonstrates the creation of Singularity containers using Cloud Build.

It supports the [Tutorials in Cloud Shell](https://cloud.google.com/shell/docs/tutorials) mechanism. To test it that way (preferred)

```
git clone https://github.com/wkharold/community
cd community
git checkout singularity-containers-with-cloud-build
cd tutorials/singularity-containers-with-cloud-build
teachme index.md
```

wardharold committed:
Merge remote-tracking branch 'upstream/master'

wardharold committed:
Initial commit

wardharold committed:
Merge remote-tracking branch 'upstream/master'

wardharold committed:
Merge branch 'master' into singularity-containers-with-cloud-build

wardharold committed:
Draft complete

wardharold committed:
Merge branch 'master' into singularity-containers-with-cloud-build

johnlabarge committed:
cloudbuild-angular-universal: remove heredoc that confuses HTML sanitizer (#649)

hongalex committed:
IoT RPi Gateways tutorial update (#650)

* iot-rpi: clarify python version and add device-specific requirements.txt

* iot-rpi: fix list numbering

* iot-rpi: fix typos

wkharold committed:
Merge remote-tracking branch 'upstream/master'

grant committed:
Format all code snippets in md files. (#640) (#642)

wkharold committed:
Using slurm to host jupyter notebooks review (#651)

* Initial commit

* Fix cloud shell link

* Fix cloud shell link

* Add frontmatter

* Add date_published

* Incorporate review feedback

Update index.md (2) provision -> provisioner

Update index.md (3) capitalize 'Helm'

Update index.md (4) add apostrophe

Update index.md (6) consistent capitalization of NFS-Client Provisioner

Update index.md (7) 'instance verify' -> 'instance to verify'

Update index.md (8) add periods

Update index.md (9) Gb -> GB

* production pass

I changed the markdown for code blocks from ``` delimiters to indentation-style syntax, since the latter works more consistently on the community site. I also updated a link to point to the GCP repo, not wkh's personal repo.

* Couple more cleanup fixes

* setup for resync

* manual sync hack

* Community review commit

* Fix activate switch in gcloud command

* copy-editing

* first copy-edit pass

I did a first, quick copy-edit while replacing image links and other links.

A closer edit will follow soon.

* some more formatting cleanup

ToddKopriva committed:
fixed link and publication date (#652)

wkharold committed:
Update index.md (#656)

Fix broken Cloud Shell button link
Add missing cluster verification command

billyjacobson committed:
Update cbt-import-csv.md (#655)

johnlabarge commented on tutorials/singularity-containers-with-cloud-build/index.md:
```diff
@@ -0,0 +1,156 @@
+author: wardharold
+tags: Cloud Build, Singularity, HPC
+date_published: 2019-01-18
+---
```
I think it would be good to have a diagram here.  Make a simple crappy one and then ask the graphics team to fix it. They are awesome.

johnlabarge commented on tutorials/singularity-containers-with-cloud-build/index.md:
````diff
@@ -0,0 +1,156 @@
+
+Once the build completes verify that the container with created using the command.
+```bash
+gsutil ls gs://${PROJECT_ID-singularity/julia-centos.sif
````
missing end }

wkharold replied:
fixed

nicoqh committed:
remove unused STORAGE_DIR env variable on Laravel flexible tutorial (#657)

jabubake committed:
Updating to latest Jackson databind library (#673)

dcecile committed:
Insert missing code close marks (#674)

jscud committed:
Adds a quickstart for Go in App Engine. (#612)

* Adds a quickstart for Go in App Engine.

* Add metadata and link to open in Cloud Console.

* Renaming Go GAE Quickstart to fit filename pattern

* restructured to Conclusion section

I restructured the "Conclusion" section so that it would have a reasonable layout without relying on the special cards.

I'll let Jeff tell me if this messes things up on the Neos side.

ToddKopriva committed:
UI orientation for non-interactive case (#675)

ToddKopriva committed:
Update go-gae-quickstart.md (#676)

thezjy committed:
Update Dockerfile (#677)

Remove an extra "\", which causes error.

jscud committed:
Update go-gae-quickstart with new default URL (#678)

* Adds a quickstart for Go in App Engine.

* Add metadata and link to open in Cloud Console.

* Renaming Go GAE Quickstart to fit filename pattern

* restructured to Conclusion section

I restructured the "Conclusion" section so that it would have a reasonable layout without relying on the special cards.

I'll let Jeff tell me if this messes things up on the Neos side.

* Update default URL for deployed Go GAE app.

wardharold committed:
Merge branch 'master' of https://github.com/wkharold/community

wardharold committed:
Merge remote-tracking branch 'upstream/master'

wardharold committed:
Add missing '}'

wardharold committed:
Split the Cloud Shell specific file out and refer to it in the button

ToddKopriva committed:
moving tutorial to archived folder (#682)

This tutorial is out of date, so it is being archived.

To republish this tutorial, submit a pull request that updates the tutorial and moves it to the /community/tutorials directory.

ToddKopriva committed:
archiving real-time-bidder tutorial (#686)

* adding status readme for archived tutorial

* moving from tutorials folder to archived folder

* Delete real-time-bidder-solution-for-google-cloud-platform_image_0.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_1.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_2.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_3.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_4.png

* more detail about what is out of date

vsoch committed:
adding start of tweaks to index.md, and updating Dockerfile with newer singularity release

hongalex committed:
IoT Gateways: typo with the cd into repo step (#688)

* IoT Gateways: typo with the cd into repo step 

fixed minor typo in the step that changes into this tutorial's directory

* fix another instance of incorrect directory name

vsoch committed:
renaming index.md to README, installing singularity release over branch

vsoch committed:
adding debugging note

wardharold committed:
index.md is now README.md

vsoch committed:
adding option to pull or use compute

vsoch committed:
cleaning up spacing in julia.def

vsoch committed:
finishing up readme notes

vsoch committed:
Merge branch 'singularity-containers-with-cloud-build' into singularity-containers-with-cloud-build-vsoch-updates

vsoch committed:
renaming the builder configuration to builder.yaml and making the two distinct in the README

vsoch committed:
Merge branch 'singularity-containers-with-cloud-build-vsoch-updates' of github.com:vsoch/google-cloud-community into singularity-containers-with-cloud-build-vsoch-updates

Civil committed:
Add a Strongswan <-> Cloud VPN interop guide (#672)

* Add a Strongswan <-> Cloud VPN interop guide

This commit adds an interoperability guide between strongswan and
Cloud VPN. Guide covers Polcy-based VPN and Dynamic (BGP).

It was tested to work on a test project that simulates On-Prem with
Strongswan, but based on GCP.

* Fix "author" field

This should make frontmatter happy.

* Fix "date_publised" field

Add a published date (assuming as it's today).

* Edits for grammar and style

* Fix capitalization and QA issues

* Fix capitalization of "strongSwan"
* Fix capitalization of "BIRD"
* Properly acknowledge second author
* Add a link to BIRD official website

* Add a missing letter to author's job title

* changing date to satisfy presubmit test

We can change the date to the later, actual date when we publish. I just wanted to clear this failure of a presubmit test to make sure that we are otherwise clear.

* adjusting codeblocks to use indentation

Because our publishing process deal badly with codeblocks in codefences, I converted the codeblocks to use indentation.

* just a few nits before publishing

vsoch committed:
adding to CONTRIBUTORS

bshaffer committed:
Adds wordpress community article (#687)

* adds wordpress community article

* editorial and production pass

I did a light copy-edit and fixed up a few production issues, including converting codeblocks to use indentation, since our publishing system deals badly with codefences in lists.

willgoldby committed:
Added missing colon. (#693)

willgoldby committed:
Style and usage edits. (#694)

* Added missing '$' to terminal commands.

* Added missing colons to 'note' messages.

* Eliminated third person usage per style guide (https://developers.google.com/style/person)

* Edit

* Revert "Eliminated third person usage per style guide (https://developers.goo…"

* Removed period and added colon.

lepistom committed:
New tutorial: Integrating Sigfox IoT Network with Google Cloud (#681)

* initial release

* removed unnecessary contrib and license

* moved images to public bucket and updated links

* part-way through edit pass

I'm pausing here for now. I still have more editing to do. I'm dealing with formatting and production issues while doing the copy-edit, so the later stages should go quickly.

* finishing editing and production pass

I fixed several small formatting and style issues.

* Update index.md

bshaffer committed:
Adds Laravel tutorial for App Engine Standard PHP 7.2 (#658)

* adds laravel app engine standard sample

markdown fixes

markdown fixes

markdown fixes

markdown fixes

markdown testing

markdown testing

markdown testing

markdown testing

* Update index.md

* fixes missing code block

* production pass

converting codeblocks to use indentation required by our quirky publishing process

bshaffer committed:
Adds symfony on App Engine Standard tutorial (#695)

* adds symfony on app engine standard tutorial

* typo fix

bshaffer committed:
Mark php55 symfony tutorial as deprecated (#696)

* Mark php55 symfony tutorial as deprecated

* Update symfony-hello-world-app-engine.md

* fixing title for branding guidelines

bshaffer committed:
Minor laravel improvements (#697)

* Misc laravel improvements

* Update index.md

* Update index.md

ToddKopriva committed:
edit and production pass for App Engine Symfony tutorial (#698)

* adding status readme for archived tutorial

* moving from tutorials folder to archived folder

* Delete real-time-bidder-solution-for-google-cloud-platform_image_0.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_1.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_2.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_3.png

* Delete real-time-bidder-solution-for-google-cloud-platform_image_4.png

* more detail about what is out of date

* editing/production pass on tutorial merged too soon

* removing image

ToddKopriva committed:
fixing incorrect indentation that led to bad numbering (#699)

bshaffer committed:
Adds bug fix for laravel deployment (#700)

bshaffer committed:
Fixes whitespace for codeblocks in laravel (#701)

* Fixes whitespace for codeblocks in laravel

* Update index.md

ToddKopriva committed:
fixing indentation and numbering for publishing (#702)

* fixing indentation and numbering for publishing

* Update index.md

ToddKopriva committed:
fixed notes that appeared as code (#703)

bshaffer committed:
Wordpress tutorial improvements (#707)

* adds a more clear line for finding out your DB region

* adds note about debugging

* adds links to added files and provides better information concerning wp-config.php

* simplifies local Cloud SQL proxy commands

* removes blockquotes because they're not parsed

* Update index.md

* copy-edits, removal of unused block quote markers

bshaffer committed:
Fixes symfony tutorial (#709)

* Fixes symfony tutorial

Somehow the doctrine section of this was not updated from the flex tutorial. This is now fixed.

* making 'Note' consistent, fixing indentation

Kcr19 committed:
Fix for Issue #628 (#710)

* Updates to repo

* Update index.md

* Update index.md

* Update index.md

Fixed typos

* Update readme.md

Updates based on review feedback

* Formatting fixes for publishing

* Fix #628

* editorial and production pass

arieljassan committed:
Running Partial Redaction of Credit Card Images with Google Cloud Functions and Data Loss Prevention API (#680)

* Setting Up a Standalone Tensorflow Prediction Device with Raspberry Pi and Google Cloud

added Setting Up a Standalone Tensorflow Prediction Device with Raspberry Pi and Google Cloud

* minor

* publishing details added

* edition of frontmatter

* frontmatter issue

* frontmatter issue

* frontmatter issue

* frontmatter issue

* frontmatter issue

* frontmatter issue

* frontmatter issue

* frontmatter issue

* Update index.md

* Update tf_server.py

* Update index.md

* Update index.md

* Update index.md

* Update index.md

* production pass

- moved images to /community bucket
- converted codeblocks to use indentation, not codefences with three backticks
- light copy-edit

* Add files via upload

* Update config.json

* fixing typos and branding issues

* copy-edit and fixing indentation

* updating image links

* Add files via upload

* Update main.py

correction to the 'import' instruction (line 14)

* update image file name

* delete images/credit-card1.jpg

Commit:
new tutorial: Load testing an IOT Application Using GCP and Locust (#706)

* added tutorial load-testing-iot-using-gcp-and-locust

* index.md changes

* fix linter errore

* fix linter errore

* fix linter errore

* fix linter errore

* fix links to images

* copy-edits, branding

* fixed typo

* copy-edit

* copy-edit

* copy-edit and production pass

This copy-edit includes many corrections for branding purposes, style, and structure.

I also changed codeblocks to use the indentation style required by our quirky publishing process.

* fixing author name to be GitHub handle

ToddKopriva committed:
fixing title capitalization (#711)

ToddKopriva committed:
links, another typo (#712)

vsoch committed:
requested changes, removing steps and references to we

wkharold committed:
Merge pull request #5 from vsoch/singularity-containers-with-cloud-build-vsoch-updates

Singularity containers with cloud build updates

wardharold committed:
Merge remote-tracking branch 'upstream/master'

wardharold committed:
Merge branch 'master' into singularity-containers-with-cloud-build

vsoch commented:
holy cow, that's a ton of commits! This is awesome!

ToddKopriva committed:
fixing codeblock that rendered as body text (#720)

* fixing codeblock that rendered as body text

* Update run-hapijs-on-google-app-engine.md

* Update run-hapijs-on-google-app-engine.md

ToddKopriva committed:
fixing codeblock that appeared as regular text (#721)

* fixing codeblock that appeared as regular text

This converts a codeblock to use the indentation style required by our quirky publishing system. I also fixed a couple of other minor issues.

* Update how-to-roll-your-app-engine-managed-vms-app-back-to-a-previous-version-part-1.md

* Update how-to-roll-your-app-engine-managed-vms-app-back-to-a-previous-version-part-1.md

* Update how-to-roll-your-app-engine-managed-vms-app-back-to-a-previous-version-part-1.md

* Update how-to-roll-your-app-engine-managed-vms-app-back-to-a-previous-version-part-1.md

ToddKopriva committed:
fixing codeblock that appeared as plain text (#722)

* fixing codeblock that appeared as plain text

This converts codeblocks to use the indentation style required by our quirky publishing system. I also fixed a couple of other minor issues.

* Update install-bower-dependencies-on-google-app-engine.md

* Update install-bower-dependencies-on-google-app-engine.md

ToddKopriva committed:
fixing codeblocks that rendered as body text (#723)

* fixing codeblocks that rendered as body text

This converts codeblocks to use the indentation style required by our quirky publishing system. I also fixed a couple of other minor issues.

* Update ghost-on-app-engine-part-2-monitoring.md

bshaffer committed:
Fixes typo in wordpress command (#725)

ToddKopriva committed:
fixing codeblocks that appeared as body text (#724)

* fixing codeblocks that appeared as body text

This converts codeblocks to use the indentation style required by our quirky publishing system. I also fixed a couple of other minor issues.

* Branding and dead link fixes

* Update run-meteor-on-google-app-engine.md

ToddKopriva committed:
fixing some more codeblocks that appeared as plain text (#726)

* fixing codeblocks that rendered as plain text

* Update index.md

* Update index.md

* Update nginx-reverse-proxy-docker.md

* Update index.md

* Update index.md

* Update index.md

* Update nginx-reverse-proxy-docker.md

* Update index.md

* Update index.md

* Update index.md

* Update index.md

* Update appengine-ruby-rails-activejob-sidekiq.md

ToddKopriva committed:
fixing badly formatted codeblocks and branding in titles (#727)

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixes codeblocks, branding, and issue 715

fixes issue #715

* fixing badly formatted codeblocks

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* Update index.md

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* Update index.md

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* Update index.md

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* Update index.md

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblock

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* fixing badly formatted codeblocks and branding in titles

* Update index.md

* Update kotlin-springboot-container-engine.md

* Update appengine-ruby-rails-activejob-pubsub.md

* Update appengine-ruby-rails-activejob-pubsub.md

* Update index.md

* Update index.md

* Update index.md

* Update index.md

* Update index.md

* Update index.md

* Update index.md

* Update send-email-with-sparkpost-and-nodejs-on-google-app-engine.md

* Update index.md

* Update zero-to-deploy-with-chef.md

* Update index.md

* Update index.md

* Update index.md

wardharold committed:
Style edits

wardharold committed:
Merge remote-tracking branch 'upstream/master'

wardharold committed:
Merge branch 'master' into singularity-containers-with-cloud-build

wardharold committed:
Sync with README

wardharold committed:
Add Overview diagram

wkharold committed:
Update README.md

vsoch commented on tutorials/singularity-containers-with-cloud-build/README.md:
```diff
@@ -7,7 +7,9 @@ date_published: 2019-01-18
-In constrast to [Docker](https://www.docker.com/), the Singularity container binary is designed specifically for HPC workloads. 
+In constrast to [Docker](https://www.docker.com/), the Singularity container binary is designed specifically for HPC workloads.
+
+This diagram illustrates the process ![diagram](overview.png)
```
Wow these diagrams are beautiful!

For the cloudbuild.yaml, are you able to tap him over a little to the left? So the text isn't broken?

![image](https://user-images.githubusercontent.com/814322/52666445-2d003700-2ec3-11e9-9493-8396192e7e99.png)

wkharold replied:
Nice catch. Done.

vsoch replied:
Looks good!

wardharold committed:
Fix markdown for Cloud Shell tutorial interpreter

wardharold committed:
Fix bad text wrap

wardharold committed:
Use the right quotes
